examples/particles.c

```diff
@@ -457,7 +457,12 @@ static void draw_particles(GLFWwindow* window, double t, float dt)
     {
         struct timespec ts;
         clock_gettime(CLOCK_REALTIME, &ts);
-        ts.tv_nsec += 100000000;
+        ts.tv_nsec += 100 * 1000 * 1000;
+        if (ts.tv_nsec >= 1000 * 1000 * 1000)
```

Comment:
@elmindreda 
Quoting the reference http://en.cppreference.com/w/c/thread/thrd_errors :

    enum {
        thrd_success = /* unspecified */,
        thrd_nomem = /* unspecified */,
        thrd_timedout = /* unspecified */,
        thrd_busy = /* unspecified */,
        thrd_error = /* unspecified */
    };

Therefore, the following [defines](https://github.com/tinycthread/tinycthread/blob/b8a4c7d4ddd1e8ec456d87e24bbe3b28b24ae343/source/tinycthread.h#L170) in TinyCThread look very suspicious to me (especially since success is normally always defined as 0 in POSIX!): 

    /* Function return values */
    #define thrd_error    0 /**< The requested operation failed */
    #define thrd_success  1 /**< The requested operation succeeded */
    #define thrd_timedout 2 /**< The time specified in the call was reached without acquiring the requested resource */
    #define thrd_busy     3 /**< The requested operation failed because a tesource requested by a test and return function is already in use */
    #define thrd_nomem 4 /**< The requested operation failed because it was unable to allocate memory */`

Anyway, `cnd_timedwait()` returned `thrd_error` when `ts` was invalid (I'm testing on Linux), so apart from the weird defines, TinyCThread is working as it should.
Therefore I don't see what you mean with:

> The fix should probably be upstreamed to TinyCThread.